tests/unit/CardUtilsTest.ts

```diff
@@ -9,44 +13,194 @@ const longDateHyphen = '09-2024';
+        });
+
+        it('Should return custom feeds only since direct feeds are not exist', () => {
+            const companyFeeds = CardUtils.getCompanyFeeds(cardFeedsCollection.FAKE_ID_3);
+            expect(companyFeeds).toStrictEqual(customFeeds);
+        });
+
+        it('Should return empty object if undefined is passed', () => {
+            const companyFeeds = CardUtils.getCompanyFeeds(undefined);
+            expect(companyFeeds).toStrictEqual({});
+        });
     });
 
-    it('Test MMYY format for getting expirationDate month and year', () => {
-        expect(cardUtils.getMonthFromExpirationDateString(shortDate)).toBe(expectedMonth);
-        expect(cardUtils.getYearFromExpirationDateString(shortDate)).toBe(expectedYear);
+    describe('removeExpensifyCardFromCompanyCards', () => {
+        it('Should return custom feeds without filtered out "Expensify Card" bank', () => {
+            const companyFeeds = CardUtils.removeExpensifyCardFromCompanyCards(cardFeedsCollection.FAKE_ID_3);
+            expect(companyFeeds).toStrictEqual(customFeedsWithoutExpensifyBank);
+        });
+
+        it('Should return direct feeds without any updates, since there were no "Expensify Card" bank', () => {
+            const companyFeeds = CardUtils.removeExpensifyCardFromCompanyCards(cardFeedsCollection.FAKE_ID_2);
+            expect(companyFeeds).toStrictEqual(directFeeds);
+        });
+
+        it('Should return empty object if undefined is passed', () => {
+            const companyFeeds = CardUtils.removeExpensifyCardFromCompanyCards(undefined);
+            expect(companyFeeds).toStrictEqual({});
+        });
     });
 
-    it('Test MM/YYYY format given MM/YY', () => {
-        expect(cardUtils.formatCardExpiration(shortDateSlashed)).toBe(longDateSlashed);
-        expect(cardUtils.formatCardExpiration(shortDateSlashed)).toBe(longDateSlashed);
+    describe('getSelectedFeed', () => {
+        it('Should return last selected feed', () => {
+            const lastSelectedFeed = CONST.COMPANY_CARD.FEED_BANK_NAME.VISA;
+            const selectedFeed = CardUtils.getSelectedFeed(lastSelectedFeed, cardFeedsCollection.FAKE_ID_1);
+            expect(selectedFeed).toBe(lastSelectedFeed);
+        });
+
+        it('Should return the first available feed if lastSelectedFeed is undefined', () => {
+            const lastSelectedFeed = undefined;
+            const selectedFeed = CardUtils.getSelectedFeed(lastSelectedFeed, cardFeedsCollection.FAKE_ID_1);
+            expect(selectedFeed).toBe(CONST.COMPANY_CARD.FEED_BANK_NAME.MASTER_CARD);
+        });
+
+        it('Should return undefined if lastSelectedFeed is undefined and there is no card feeds', () => {
+            const lastSelectedFeed = undefined;
+            const cardFeeds = undefined;
+            const selectedFeed = CardUtils.getSelectedFeed(lastSelectedFeed, cardFeeds);
+            expect(selectedFeed).toBe(undefined);
+        });
     });
```

Comment:
Could you also test selecting the direct feed please?

Reply:
Done https://github.com/Expensify/App/pull/52268/commits/07cb200e2800182477f89039910a56bdd486f0a6